drivers/flash/flash_shell.c

```diff
@@ -193,13 +193,13 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 		return -EINVAL;
 	}
 
-	flash_write_protection_set(flash_dev, 0);
-
 	for (uint32_t i = 0; i < size; i++) {
 		test_arr[i] = (uint8_t)i;
 	}
 
 	while (repeat--) {
+		flash_write_protection_set(flash_dev, false);
```

Comment:
Just wondering: shouldn't write protection be "restored" before (or right after) exiting the while loop?

@nvlsianpu ^^

Reply:
In that case should be restored after while loop.

In general recommendation is to set flash protecting after packet of write/erase operation was completed.

Reply:
@eugmes ^^

Reply:
@nvlsianpu I can fix this, but the original code also didn't restore write protection.

Reply:
Fixed. It's either `goto` or duplicate code.

Reply:
thanks!

Reply:
wonder if we really need to have "goto" here, isn't there a refactor that removes the need, without duplicating code too much?

Reply:
@ioannisg How about this?

Reply:
oops.

Reply:
Now.